### ISSUE TITLE
Board not powering Up

### ISSUE THREAD

**Issue by deadca763** [juckettd/RaspberryPiCM4Handheld7Inch#3]
Hello and first of all congratulations for this fabulous project!

I have a problem:
After soldering all the components on the PCB and after having assembled everything, the card refuses to start, when I press the POWER button nothing happens, I have a vBAT of 3.7v coming out but nothing more.
Nothing coming out from TR4 TR5 TR6 transistors

Any ideas?

I also saw on the Battery Charging IC that there is no connection on the vBAT and vBUS, is it normal?

![20230116_185158](https://user-images.githubusercontent.com/122821820/212744286-8922048c-1c26-4576-a6f3-c72bb3592ba2.jpg)

Thanks in advance

**Comment by juckettd** [comment]
It looks like to me that the board is missing some copper pours in those areas. I'd guess that it was a manufacturing area because those copper pours definitely exist in the CAM files. If you have enough skill with a soldering iron and have a fine enough tip you can solder wires to those same connections. The bottom one shouldn't be too difficult, but the top one might be harder as you have to connect to both of the pins to avoid running too much current through a single pin.

**Comment by juckettd** [comment]
After looking at more of the picture there are more copper pours missing as well like the pour going up to L1 is missing

**Comment by deadca763** [comment]
Indeed thanks, i've tried to solder these but things gets weird, like with the battery nothing else happens, and with the usb the rpi started to make some coil whine noises without pressing the power button, i'll re-check all solders and hope nothing bad happens !

**Comment by juckettd** [comment]
Because there are copper pours missing with the battery charging circuit, it wouldn't surprise me if there are copper pours missing other places as well. That could definitely cause problems with power and ground connections to other items. Because I used copper pours to do better heat management with the circuits. So you might have a bigger problem and you should check your board with the supplied Eagle board files to see what is missing.

**Comment by deadca763** [comment]
Okay thanks for the advice

**Comment by deadca763** [comment]
The device is alive! 
RPi is booting well and working, however it starts by itself when bat is plugged in, and can't stop it with the PWR Button, only resets it.

**Comment by juckettd** [comment]
Yes that is correct. The circuitry will auto power up when it gets power from a USB source. You have to send it a signal from the I2C port of the raspberry pi to shut down power completely. The script I wrote does this to allow you to turn off the device. This is the monitor script that gets installed by the setup script. But I would like to point out that this script uses Dispmanx to do overlays on the screen and that is not compatible with the KMS driver. You have to use the FKMS video driver in your boot/config.txt file. 

However, when it is charging it will be giving power to the rest of the system at the same time. But you can still shut down the raspberry pi even while it is charging through the script to save on some power. But the power from the device will still be there. It is like turning a Raspberry Pi off while leaving it plugged into the wall.

**Comment by deadca763** [comment]
Yes however it happens even without usb power on it, the monitor script seems to work with overlays but when i execute the script's shutdown function it is still powered up (rpi shut down but power led still ON and all 3,3v 5v tensions still here)

Secondly, i tried charging the battery with usb power on it during all night and battery wasn't full (around 80% at 3,8v)

And lastly if you can tell me how to configure the teensy to be working with emulationstation, the buttons are recognized with the command tycmd and working fine, but when in emulationstation it is stuck at "Hold a button to configure gamepad"

Thanks for your help !

**Comment by deadca763** [comment]
Here is a video of the problem when I push the power button and the controller buttons not responding in emulationstation, (it is working by monitoring with command "tycmd monitor")
https://user-images.githubusercontent.com/122821820/214675444-c1cb18a4-9dab-4a5c-907a-bc6735c6287c.mp4

**Comment by juckettd** [comment]
How did you wire up the teensy? I'm looking through the instructions on the readme right now and something that I don't think I mentioned is you have to wire up the D+ and D- connections from the teensy to the board. There are four pads underneath the teensy and 4 pads in about the same spot on the board that I made. you don't need to connect the VBUS and GND connections, but you have to connect the D+ and D- connections. I just did it with some wire. With the teensy already soldered to the board it is a little tricket to do it but you can still get to both pads on both boards with a little work. I am assuming that this is the reason why your contrroller doesn't work.

As for the power button I'm not sure the monitor script is working, because if it is then there would be icons showing up in the upper right hand portion of the screen. Also the way the device reboots is a little concerning as it shouldn't turn white like that on the screen. That means that there is data corruption on the LCD lines going to the screen. This usually happens if there is a short somewhere that is causing the device to shut itself down due to the short and the power connection. With the script not running the button shouldn't do anything unless you hold it down for about 20 sec (where it does a hard reboot). You should check the trace with the button to to see if there is a bridged connection where power is getting connected to ground when the button is pressed.

**Comment by deadca763** [comment]
I wired the 4 wires of the usb teensy, maybe i should remove the 5v and gnd as you said ?

And for the power problem i'll check for short circuit, it is clearly the origin of the problem i think lol

Edit: did removed the 5v and ground wires from the usb of the teensy and nothing changed for the controller
Do I need to install a driver of something on the pi ?

**Comment by juckettd** [comment]
You shouldn't have to disconnect the VBUS and GND connections as they are just redundant. I doubt any power loops are causing this issue. Did you push the code to the teensy?

**Comment by juckettd** [comment]
Also in the config.txt file there is a USB statement you need to put in there to enable the USB 2.0. The statement is "dtoverlay=dwc2,dr_mode=host". Make sure you also have this in your file. If the USB 2.0 still doesn't work then you should check the solder joints of the USB pins on the socket connector where the PI connects on the custom board. And check the solder joints on the USB hub circuit to make sure all of those pins are connected correctly.

**Comment by deadca763** [comment]
I think it is working because the screen displays "1 gamepad detected" when d+ and d- of the teensy are connected, I'll re-check all what you said to be sure.

For the power problem I assume this is a difficult thing,
What is happening is when I push the button, QON, GPIO7 and GLOBAL_EN are falling to 0v, I think it is normal because shorted to ground with the transistors, but should I have Voltage at RUN_PG when rpi is started?
![image](https://user-images.githubusercontent.com/122821820/214908657-886b479d-2080-4460-8023-a0022458ef59.png)

**Comment by deadca763** [comment]
Also there is the end of the config.txt file 

[pi4]
dtoverlay=vc4-fkms-v3d
max_framebuffers=2

[all]
#dtoverlay=vc4-fkms-v3d
overscan_scale=1
dtoverlay=dwc2,dr_mode=host
gpu_mem_256=128
gpu_mem_512=256
gpu_mem_1024=256
force_turbo=1
h264_freq=333
dtoverlay=hifiberry-dac
dtoverlay=i2s-mmap
gpio=11=op,dh
#ignore_lcd=1
cec_osd_name=Aegis
hdmi_blanking=1
gpio=9,10=ip,pu
gpio=7,8=ip,pu
dtoverlay=i2c1,pins_2_3
#hdmi_enable_4kp60=1
enable_tvout=1

**Comment by deadca763** [comment]
After searching, this is the TR4 transistor that makes the screen goes blank when shorting it only, there is also 5v coming from GLOBAL_EN, is it normal?

**Comment by juckettd** [comment]
There is a Buffer Circuit that feeds the T4 transistor. This buffer circuit is connected to RUN_PG. So when the PI is off (But not the power to the handheld) then RUN_PG is low, which will then pull GLOBAL_EN low. This causes the PI to turn on even if power is already feeding the device. This is similar to the idea of turning the PI back on when it was shutdown through software but not unplugged.

When the PI is currently on then RUN_PG should be high, thus connecting GLOBAL_EN to a 5V source, which will do nothing. I did it this way because pulling GLOBAL_EN low, while the PI is on, does not produce a safe shutdown. So instead the input from GPIO7 will be used to trigger a safe shutdown routine.

The fact that you are getting a reboot when pressing the power buttom most likely means that the RUN_PG signal is not connected to the buffer circuit. This is most likely caused by a bad solder join either on the socket connectors to the PI or at the buffer itself. Although it could also be caused at the connection to the transistor as well so I'd check that too. I have included a picture of the connection on the buffer circuit you need to check and trace to see what it is. When the PI is off it should be GND. When the PI Is on it should be 3.3V (which the buffer circuit ups to 5 for a safe connection to GLOBAL_EN)

![image](https://user-images.githubusercontent.com/24655543/214944802-60839ea3-72e2-4766-9068-68a5eeb608ce.png)

**Comment by juckettd** [comment]
For the controller still not working, I agree that your teensy seems to be working correctly already. Sorry I didn't see that it recongized the controller already. But since the buttons don't work it is most likely a connection issue from the Wii U button connector to the circuit board. I have included a picture for the connector to check the following things (also check the other side as well not shown in the picture):

Make sure all the pins are soldered cleanly
Make sure there are no breaks in the flex strip that goes into the connector
Make sure the flex strip is sitting correctly in the connector
Lastly, make sure there are no bare spots in the flex strip rigth where the connector connects (I have found that the flex strip wears out quickly and if I plugged/unplugged the strip to many times it caused a bare spot and the connector no longer made a clean connection and I got intermitten connection issues).

![image](https://user-images.githubusercontent.com/24655543/214946165-a4c2b975-7361-476e-a4ce-5a9d11546944.png)

**Comment by deadca763** [comment]
Ok the power problem is fixed! perfectly working now, it was a bad solder at the RUN_PG pin on the socket connector for the pi...
The power button now works for reboot, shutdown, and power up!
Thanks alot for the help!

Now let's go for the controller problem

Edit: Just 1 problem remains, when charging and shutting down the board to let it charge gently, it won't boot up again unless disconnect the battery and plug it again

**Comment by deadca763** [comment]
Here is a video of the controller recognized with the command tycmd
https://user-images.githubusercontent.com/122821820/214992140-0d95a97e-cc42-4b81-a0e8-df9a5a583cc9.mp4
Don't know why it isn't working with emulationstation

**Comment by juckettd** [comment]
Awesome that the power button is now working. For the the USB controller though, it seems like the connection is just fine based on your tycmd demo. I'm guessing that maybe XpadNeo isn't installed. I believe in my version of retropie that it was already installed but maybe there is a chance it wasn't installed for you. I think you can safely install both Xpad as well incase Xpadneo doesn't work. You can do this through the retropie setup script with a keyboard and mouse. Or you can do it through a remote client.

**Comment by deadca763** [comment]
Okay let's try this so !

And do you know if it's possible to charge the device when it's powered off ?

**Comment by deadca763** [comment]
Tried install xpadneo (xpad was already installed) and still no changes
Do I need to do something on the arduino software before compiling your code?
I'm configured like this : 
Board Type : Teensy LC
USB Type : Keyboard
CPU Speed : 48MHz

**Comment by deadca763** [comment]
Okay working now! it was the Xinput library missing on my arduino software x)

Problems still remaining : 
-Hdmi plug not working
-Charging when powered off

**Comment by juckettd** [comment]
That is great that the controller is working. I didn’t think to check the teensy libraries as I assumed it wouldn’t even install without the xinput library. When I update the install instructions I need to make sure that is specified. 

The charging while on is unfortunately a side effect of the way it was designed. I modeled it after something like the iPhone where it is on when it charges. However, you can turn the raspberry pi off while it is charging once you plug the power cord in. But this still leaves power running to the pi. It is like doing a software shutdown without unplugging a regular PI 4. So it saves some power while charging but not everything.

The HDMI not working is probably due to some bridged or unsoldered pins on the HDMI connector. Those pins are very close together and I found I had to clean them up after I put it together using my reflow station. A fine soldering tip comes in a handy here. If that isn’t the case for you then I’d also check the pins on the PI socket connected to see if something similar is happening.

**Comment by deadca763** [comment]
Okay no problem for the charge
I'll re-check all solders on the hdmi port
I tried the device with a little game and working fine! but how the fan is working? Because it stays off, do I need to put the thermister on the rpi cpu or somewhere?

**Comment by juckettd** [comment]
Just to give you info on the charging thing, I realized after I used the device for a while that the way the power and charging works isn't that great. I am changing it for a future design but that will be for an Aegis 2 device and not this one unfortunately. However, I do have a update that I will be pushing up soon for this that will address some software issues where it lags a bit in some games with the screen overlay. I also made some really minor PCB changes that improved some power routing issues to the SD card. But that only shows up when the power gets below 3.0V from the battery. I didn't notice it until recently because I never let the battery get that low.

**Comment by juckettd** [comment]
Also for the fan. it is setup in the monitor script to turn on when the Raspberry PI reaches 50 degress C. So if you play a intense game you will hear it come on from time to time. But otherwise it will stay off. And when it does come on it will only come on for as long as it takes to drop the temp back down below 50 and then another minute after that to prevent it from turning back on immediately again.

**Comment by deadca763** [comment]
Can't wait to see the ver. 2 finished!
For the fan, the RFB4505 reference no longer available, I replaced it with a 30x30x4mm square fan, here https://fr.aliexpress.com/item/1005003167391047.html?spm=a2g0o.order_list.order_list_main.11.5b745e5bsRdVYG&gatewayAdapt=glo2fra
Still waiting to receive it i replaced temporarily with a little dc fan to see if working but nope, I checked the Voltages and there is around 3v output from the + and - pins of the connector of the board, Don't know if it's normal but seems not.

**Comment by juckettd** [comment]
The voltage to the fan is 3.3V. I run it lower then 5V because at 5V the original fan was just too loud and had a high pitched squeal to it. I tested it out a lot to make sure it can handle 3.3V to run at a lower speed. But not all fans can accomodate a lower voltage then it was originally designed for.

**Comment by deadca763** [comment]
Yesterday a weird thing happened, the battery was low and the auto stop function didn't work, Bat came below 3v (2.6v) and device started to disfunction in loop, like screen blanking, restart by itself and unability to shutdown, I had to disconnect the battery to stop it and remove raspberry to charge the battery without starting the device. Don't know what happened :/

A little question, can I run the device with 2 systems? like the retropie on the eMMC of the Pi and another system on the SD Card?

**Comment by Musicislife6984** [comment]
> The device is alive! 
> 
> RPi is booting well and working, however it starts by itself when bat is plugged in, and can't stop it with the PWR Button, only resets it.

Did you find any other missing pours? I just got my PCB from JLCPCB and it is missing those as well. I appreciate any help as I am getting ready to start my build.

**Comment by deadca763** [comment]
All missing pours concern the bat_chr ic, 5v_reg and the audio_dac, on the board file you can see all of it by the dotted lines, some screens here to show you.

PS: you also need to check where the 2nd little board connects to the main board, 5v alim is missing too

![Puce-charge](https://user-images.githubusercontent.com/122821820/218643116-6fc8271b-d724-49e2-8223-35a72901030f.png)
![Reg-5V](https://user-images.githubusercontent.com/122821820/218643119-7044b945-c5df-4513-803b-aedd18a92625.png)
![Audio](https://user-images.githubusercontent.com/122821820/218643122-26825622-46e9-45b2-9421-436e76733a10.png)
